src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -46,108 +54,155 @@
     <h1>Activity Feed</h1>
     <hr/>
 
-    <div style="background-color:Silver">
       <!-- Add to activity feed when:
               - Users register -- "blahblah joined!" 
               - Users creating conversations -- "blahblah created a new conversation: Convo17"
-              - Users sending messages -- Blahblah sent a message to Convo 17: Yoyo! -->
-
-
-      <!-- USER REGISTERED -->
-      <h3>New Users</h3>
-
-        <%
-        // pull new user
-        User tempUser = (User)request.getAttribute("newTempUser");
-        %>
-        <ul class="mdl-list">
-          <li>
-           <strong> <%= tempUser.getCreationTime() %>: </strong> 
-            <%= tempUser.getName() %> joined!
-          </li> 
-        </ul>
-
-     <!-- <%
-      // Pull user data
-      List<User> newUsers = (List<User>) request.getAttribute("newUsers");
-      ListIterator<User> itrU = newUsers.listIterator(newUsers.size());
-
-      // empty?
-      if(newUsers == null || newUsers.isEmpty()){
-      %>
-        <p>No New Users.</p>
-      <%
-      }
-      else{ // not empty? make a list
-      %>
-        <ul class="mdl-list">
-      <%
-        while(itrU.hasPrevious()){
-          User currUser = itrU.previous();
-      %>
-        <li>
-          <strong> <%= currUser.getCreationTime() %>: </strong> 
-          <%= currUser.getName() %> joined!
-        </li> 
-      <%
-        }
-      %>
-        </ul>
+              - Users sending messages -- Blahblah sent a message to Convo 17: Yoyo! 
+      -->
+
       <%
-      }
-      %> -->
+        // Stores
+          UserStore userStore = (UserStore) request.getAttribute("users");
+          ConversationStore convoStore = (ConversationStore) request.getAttribute("conversations");
 
+        // Lists
+          List<User> users = userStore.getLatestUsers();
+          List<Conversation> conversations = (List<Conversation>) convoStore.getLatestConversations();
+          List<Message> messages = (List<Message>) request.getAttribute("messages");
 
+        // Master List
 
+          /*
+            Overall Format:
+              Date | Mode | Username | Convo | Message Content
 
+            Convo Format:
+              Date | c | Username | Convo | NULL
 
+            User Format:
+              Date | u | Username |  NULL | NULL
 
+            Message Format:
+              Date | m | Username | Convo | Message Content
+          */
 
+          int totalSize = conversations.size() + users.size() + messages.size();
```

Comment:
I'd prefer it if you could move most of this logic to the servlet. These templates are difficult to test, but the servlet is not, so most of your business logic should be in there instead. Templates should generally only contain very simple logic that just renders previously processed data into the page.